otx/cli/manager/config_manager.py

```diff
@@ -243,6 +241,96 @@ def auto_task_detection(self, data_roots: str) -> str:
         self.data_format = self.dataset_manager.get_data_format(data_roots)
         return self._get_task_type_from_data_format(self.data_format)
 
+    def _configure_train_type(self):
+        """Auto train type detection.
+
+        If self.args.train_type is not None -> use args.train_type
+        If train_data_roots contains only set of images -> Self-SL
+        If unlabeled-data-roots were passed to CLI -> use Semi-SL
+        If unlabeled_images presented in dataset structure and it is sufficient to start Semi-SL -> Semi-SL
+        Overwise set Incremental training type.
+        """
+
+        def _check_is_only_images(dir):
+            """Check if a directory contains only images."""
+            import glob
+
+            pngs = glob.glob(f"{dir}/*.png")
+            jpgs = glob.glob(f"{dir}/*.jpg")
+            return len(pngs) > 0 or len(jpgs) > 0
+
+        def _check_semisl_requirements(train_dir, unlabeled_dir, thershold=0.07):
+            """Check if quantity of unlabeled images is sufficient for Semi-SL learning."""
+            all_unlabeled_images = len(os.listdir(unlabeled_dir))
```

Comment:
it would be possible to read a file called unlabeled-image-list, but that doesn't seem to be considered.
Also, this is assuming that the unlabeled data folder doesn't have subdirectories - I know that OTX currently load all images in subdirectories as well.
@sungmanc If you have any additional comments, please leave them below.

Reply:
1) I added a check before passing to that function that unlabeled_dir is a dir, not a file
2) Yes, current implementation considered just a dir with images, @sungmanc will Datumaro scan directories recursively if there are some in the tree?

Reply:
Agree, it will be again impossible to distinguish two tasks -> let's make it simple and support only one level of hierarchy. It should be written in documentation as well